src/tests/ftest/deployment/target_failure.py

```diff
@@ -0,0 +1,417 @@
+    """Test class Description: Verify target failure is properly handled and recovered.
+
+    :avocado: recursive
+    """
+    def run_ior_report_error(self, results, job_num, file_name, oclass, pool, container):
+        """Run IOR command and store the results to results dictionary.
+
+        Create a new IorCommand object instead of using the one in IorTestBase because
+        we'll run a test that runs multiple IOR processes at the same time.
+
+        Args:
+            results (dict): A dictionary object to store the ior metrics
+            job_num (int): Assigned job number
+            file_name (str): File name used for self.ior_cmd.test_file.
+            oclass (str): Value for dfs_oclass and dfs_dir_oclass.
+            pool (TestPool): Pool to run IOR.
+            container (TestContainer): Container to run IOR.
+        """
+        # Update the object class depending on the test case.
+        ior_cmd = IorCommand()
+        ior_cmd.get_params(self)
+        ior_cmd.dfs_oclass.update(oclass)
+        ior_cmd.dfs_dir_oclass.update(oclass)
+
+        # Standard IOR prep sequence.
+        ior_cmd.set_daos_params(self.server_group, pool, container.uuid)
+        testfile = os.path.join("/", file_name)
+        ior_cmd.test_file.update(testfile)
+
+        manager = get_job_manager(
+            test=self, class_name="Mpirun", job=ior_cmd, subprocess=self.subprocess,
+            mpi_type="mpich")
+        manager.assign_hosts(
+            self.hostlist_clients, self.workdir, self.hostfile_clients_slots)
+
+        # Run the command.
+        try:
+            self.log.info("--- IOR command %d start ---", job_num)
+            ior_output = manager.run()
+            results[job_num] = [True]
+            # For debugging.
+            results[job_num].extend(IorCommand.get_ior_metrics(ior_output))
+            # Command worked, but append the error message if any.
+            results[job_num].append(ior_output.stderr_text)
+            self.log.info("--- IOR command %d end ---", job_num)
+        except CommandFailure as error:
+            self.log.info("--- IOR command %d failed ---", job_num)
+            results[job_num] = [False, "IOR failed: {}".format(error)]
+
+    @staticmethod
+    def check_container_health(container, expected_health):
+        """Check container property's Health field by calling daos container get-prop.
+
+        Args:
+            container (TestContainer): Container to call get-prop.
+            expected_health (str): Expected value in the Health field.
+
+        Returns:
+            bool: True if expected_health matches the one obtained from get-prop.
+
+        """
+        actual_health = None
+
+        output = container.get_prop()
+        for cont_prop in output["response"]:
+            if cont_prop["name"] == "status":
+                actual_health = cont_prop["value"]
+                break
+
+        return actual_health == expected_health
```

Comment:
Minor/optional - I personally like this style, though there's no need to change :)
```suggestion
        output = container.get_prop()
        for cont_prop in output["response"]:
            if cont_prop["name"] == "status":
                return cont_prop["value"] == expected_health

        return False
```

Reply:
Done